components/eventing-controller/pkg/backend/jetstreamv2/connection.go

```diff
@@ -0,0 +1,45 @@
+package jetstreamv2
+
+import (
+	backendnats "github.com/kyma-project/kyma/components/eventing-controller/pkg/backend/nats"
+	pkgerrors "github.com/kyma-project/kyma/components/eventing-controller/pkg/errors"
+	"github.com/nats-io/nats.go"
+)
+
+type Builder interface {
+	Build() (ConnectionInterface, error)
+}
+
+// ConnectionBuilder helps in establishing a connection to NATS.
+type ConnectionBuilder struct {
+	config backendnats.Config
+}
+
+func NewConnectionBuilder(config backendnats.Config) Builder {
+	return ConnectionBuilder{config: config}
+}
+
+// Build connects to NATS and returns the connection. If an error occurs, ErrConnect is returned.
+func (b ConnectionBuilder) Build() (ConnectionInterface, error) {
+	config := b.config
+	jsOptions := []nats.Option{
+		nats.RetryOnFailedConnect(true),
+		nats.MaxReconnects(config.MaxReconnects),
+		nats.ReconnectWait(config.ReconnectWait),
+		nats.Name("Kyma Controller"),
+	}
+	conn, err := nats.Connect(config.URL, jsOptions...)
+	if err != nil || !conn.IsConnected() {
+		return nil, pkgerrors.MakeError(ErrConnect, err)
+	}
+
+	return conn, nil
+}
+
+// ConnectionInterface is a contract for a NATS connection object.
+type ConnectionInterface interface {
```

Comment:
Okay, one last try to find better names:
Name the `ConnectionInterface` -> `Connector`
and the `Build` function `Connect`.
Let me know what you think. Other than that, this looks good to me.

Reply:
If I change the Build() method to Connect, then the `Builder` interface needs to be renamed to `Connector` :-D

The `ConnectionInterface` is simply an interface for the `nats.Conn` (nats sdk does not offer an interface)

Reply:
`jetstreamv2.Connection`

Reply:
Never mind, maybe I am making it too complicated.